pkg/telemetry/exporters/nats.go

```diff
@@ -145,186 +132,212 @@ func NewNATSSpanExporter(ctx context.Context, opts ...NatsExporterOpts) (trace.S
 	exporter := &natsSpanExporter{
 		conn:    conn,
 		streams: expOpts.streams,
-		dlc:     make(chan *runv2.Span, expOpts.dlcBuffer),
+		buf:     newNatsBuffer(),
 	}
 
-	for i := 0; i < expOpts.dlConcurrency; i++ {
-		go exporter.handleFailedExports(ctx)
-	}
+	go exporter.handleBuffered(ctx)
 
 	return exporter, nil
 }
 
-func (e *natsSpanExporter) handleFailedExports(ctx context.Context) {
-	js, err := e.conn.JSConn()
-	if err != nil {
-		logger.StdlibLogger(ctx).Error("error access Jetstream connection")
-
-		return
-	}
+func (e *natsSpanExporter) handleBuffered(ctx context.Context) {
+	ticker := time.NewTicker(500 * time.Millisecond)
```

Comment:
instead of taking things from the channel as fast as it can, it periodically checks the buffer and attempt to ship things out that are in it.